a10_neutron_lbaas/plumbing_hooks.py

```diff
@@ -111,19 +112,57 @@ def select_device(self, tenant_id):
 
 class VThunderPlumbingHooks(PlumbingHooks):
 
-    def select_device(self, tenant_id, db_session=None):
+    def select_device_with_lbaas_obj(self, tenant_id, a10_context, lbaas_obj,
+                                     db_session=None):
         if not self.driver.config.get('use_database'):
             raise ex.RequiresDatabase('vThunder orchestration requires use_database=True')
 
+        # If we already have a vThunder, use it.
+
+        if hasattr(lbaas_obj, 'root_loadbalancer'):
+            # lbaas v2
+            root_id = lbaas_obj.root_loadbalancer.id
+            slb = models.A10SLB.find_by_loadbalancer_id(root_id, db_session=db_session)
+            if slb is not None:
+                d = self.driver.config.get(slb.device_name, db_session=db_session)
+                if d is None:
+                    raise ex.InstanceMissing(
+                        'A10 instance mapped to loadbalancer_id %s is not present in db; '
+                        'add it back to config or migrate loadbalancers' % root_id)
+                return d
+        else:
```

Comment:
This is a little inflexible. What if professional services needs to do device per tenant for lbaasv2?